store/pruning/manager.go

```diff
@@ -74,28 +76,52 @@ func (m *Manager) HandleSnapshotHeight(height int64) {
 		return
 	}
 
+	if m.snapshotInterval == 0 || uint64(height)%m.snapshotInterval != 0 {
+		// ensure that we keep only heights that match the current snap interval
+		return
+	}
+
 	m.pruneSnapshotHeightsMx.Lock()
 	defer m.pruneSnapshotHeightsMx.Unlock()
 
 	m.logger.Debug("HandleSnapshotHeight", "height", height)
 	m.pruneSnapshotHeights = append(m.pruneSnapshotHeights, height)
 	sort.Slice(m.pruneSnapshotHeights, func(i, j int) bool { return m.pruneSnapshotHeights[i] < m.pruneSnapshotHeights[j] })
+
+	if m.initFromDB && len(m.pruneSnapshotHeights) > 2 { // received 2 fresh snap heights as base level
+		m.pruneSnapshotHeights = m.pruneSnapshotHeights[1:] // drop db state element
+		m.initFromDB = false
+	}
+
 	k := 1
 	for ; k < len(m.pruneSnapshotHeights); k++ {
+		// any one missing in the sequence means they are still in flight
 		if m.pruneSnapshotHeights[k] != m.pruneSnapshotHeights[k-1]+int64(m.snapshotInterval) {
 			break
 		}
 	}
+	// compact the heights list when in-flight snapshots have landed
 	m.pruneSnapshotHeights = m.pruneSnapshotHeights[k-1:]
 
-	// flush the updates to disk so that they are not lost if crash happens.
-	if err := m.db.SetSync(pruneSnapshotHeightsKey, int64SliceToBytes(m.pruneSnapshotHeights)); err != nil {
+	if n := len(m.pruneSnapshotHeights); n > 100 {
```

Comment:
I added some logging to make this problem more visible.
The `100` is just an arbitrary number. In a normal process, the snapshot heights should not pile up.